qutip/solver/dysolve_propagator.py

```diff
@@ -0,0 +1,473 @@
+from qutip import Qobj, qeye_like
+from numpy.typing import ArrayLike
+import numpy as np
+import scipy as sp
+from numbers import Number
+import itertools
+from scipy.special import factorial
+
+__all__ = ['DysolvePropagator', 'dysolve_propagator']
+
+
+class DysolvePropagator:
+    """
+    A generator of propagator using Dysolve.
+    https://arxiv.org/abs/2012.09282
+
+    Parameters
+    ----------
+    H_0 : Qobj
+        The base hamiltonian of the system.
+
+    X : Qobj
+        A cosine perturbation applied on the system.
+
+    omega : Number
```

Comment:
Can't be a complex number.
```suggestion
    omega : float
```